src/StaticWebAssetsSdk/Tasks/Compression/ResolveCompressedAssets.cs

```diff
@@ -233,25 +246,29 @@ private bool TryCreateCompressedAsset(ITaskItem asset, string format, out TaskIt
             Log.LogError(
                 "Unknown compression format '{0}' for '{1}'.",
                 format,
-                asset.ItemSpec);
+                asset.Identity);
             return false;
         }
 
-        var originalItemSpec = asset.GetMetadata("OriginalItemSpec");
-        var relativePath = asset.GetMetadata("RelativePath");
+        var originalItemSpec = asset.OriginalItemSpec;
+        var relativePath = asset.RelativePath;
 
         var fileName = FileHasher.GetFileHash(originalItemSpec) + fileExtension;
-        var outputRelativePath = Path.Combine(OutputPath, fileName);
+        var itemSpec = Path.GetFullPath(Path.Combine(OutputPath, fileName));
```

Comment:
This was using the relative path. StaticWebAssets item specs should always be full paths